crates/oxc_linter/src/service/runtime.rs

```diff
@@ -102,105 +191,424 @@ impl Runtime {
+                    .iter()
+                    .filter_map(|resolved_module_record| {
+                        Some(Arc::clone(&resolved_module_record.as_ref().ok()?.module_record))
+                    })
+                    .collect();
+
+                modules_by_path.insert(Arc::clone(&path), records);
+
+                for record_result in &processed_module.section_module_records {
+                    let Ok(record) = record_result.as_ref() else {
+                        continue;
+                    };
+                    for (_, dep_path) in &record.requested_module_paths {
+                        if encountered_paths.insert(Arc::clone(dep_path)) {
+                            scope.spawn({
+                                let tx_resolve_output = tx_resolve_output.clone();
+                                let dep_path = Arc::clone(dep_path);
+                                move |_| {
+                                    tx_resolve_output
+                                        .send(me.process_path(
+                                            dep_path,
+                                            check_syntax_errors,
+                                            tx_error,
+                                        ))
+                                        .unwrap();
+                                }
+                            });
+                            pending_module_count += 1;
+                        }
+                    }
+                }
+
+                module_relationships.push((
+                    Arc::clone(&path),
+                    processed_module
+                        .section_module_records
+                        .iter_mut()
+                        .filter_map(|record_result| {
+                            Some(take(&mut record_result.as_mut().ok()?.requested_module_paths))
+                        })
+                        .collect(),
+                ));
+
+                if let Some(entry_module) =
+                    EntryModule::from_processed_module(path, processed_module)
+                {
+                    entry_modules.push(entry_module);
+                }
+            }
+
+            // Writing to `loaded_modules` based on `module_relationships`
+            module_relationships.par_drain(..).for_each(|(path, requested_module_paths)| {
+                if requested_module_paths.is_empty() {
+                    return;
+                }
+                let records = &modules_by_path[&path];
+                assert_eq!(records.len(), requested_module_paths.len());
+                for (record, requested_module_paths) in
+                    records.iter().zip(requested_module_paths.into_iter())
+                {
+                    let mut loaded_modules = record.loaded_modules.write().unwrap();
+                    for (specifier, dep_path) in requested_module_paths {
+                        // TODO: revise how to store multiple sections in loaded_modules
```

Comment:
Let's figure out out the TODO, it's unlikely we'll revisit this part of code and fix it later.